GreenhouseNode/GreenhouseNode/handlers/WindowHandler.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * WindowHandler.c
+ *
+ * Created: 12/14/2021 2:27:42 PM
+ *  Author: tolde
+ */ 
+#include <ATMEGA_FreeRTOS.h>
+#include <rc_servo.h>
+#include <task.h>
+#include "WindowHandler.h"
+#include "controllers/WServo.h"
+
+EventGroupHandle_t startGroup_task;
+EventBits_t dBit;
+
+typedef struct direction{
+	int8_t direction_servo;
+	int8_t windowToDestroy;
+	}direction;
+
+void start_directionTask(void * self);
+
+directionHandler_t windowHandler_create(UBaseType_t window_task_priority, EventGroupHandle_t eventBits, EventBits_t bits)
+{
+	directionHandler_t newWindow = calloc(1, sizeof(direction));
+	
+	if((newWindow == NULL))
+	{
+		return NULL;
+	}
+	
+	initialise_servo();
+	
+	newWindow = initialise_servo();
+	
+	dBit = bits;
+	startGroup_task = eventBits;
+	
+	vTaskDelay(400);
+	
+	set_servo(newWindow);
+	
+	direction_handler_intiliase(window_task_priority, newWindow);
+	
+	return newWindow;
+}
+
+void direction_handler_intiliase(UBaseType_t directionHandler_priority, directionHandler_t self)
+{
+	xEventGroupSetBits(startGroup_task, dBit);
+
+	xTaskCreate(
+	start_directionTask()
+	, "WindowTask"
+	, configMINIMAL_STACK_SIZE + 100
+	, (void*)self
+	, directionHandler_priority
+	, NULL);
+}
+
+void start_directionTask(void* self)
```

Comment:
Is this task only supposed to run once? If so it should be deleted properly once it completes.

Reply:
this